### ISSUE TITLE
stops working whenever I open a file containing html being written by javascript

### ISSUE THREAD

**Issue by rutenio** [vincaslt/vscode-highlight-matching-tag#49]
The extension stops working whenever I open a file containing html being written by javascript with un error:

> [2018-10-24 11:12:48.433] [renderer1] [error] invalid syntax at line 195 col 132:

                                          '<input class="form-check-input" type="checkbox" disabled ' + (v.ativo == 1 ? 'checked' : '')  +
                                                                                                                                     ^: Error: invalid syntax at line 195 col 132:

                                          '<input class="form-check-input" type="checkbox" disabled ' + (v.ativo == 1 ? 'checked' : '')  +
                                                                                                                                     ^
	at Lexer.next (C:\Users\Rutenio\.vscode\extensions\vincaslt.highlight-matching-tag-0.7.0\node_modules\moo\moo.js:397:13)
	at Object.parseTags (C:\Users\Rutenio\.vscode\extensions\vincaslt.highlight-matching-tag-0.7.0\out\src\tagParser.js:79:36)
	at vscode.window.onDidChangeTextEditorSelection (C:\Users\Rutenio\.vscode\extensions\vincaslt.highlight-matching-tag-0.7.0\out\src\extension.js:86:38)
	at e.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:99:496)
	at e.$acceptEditorPropertiesChanged (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:589:930)
	at t._doInvokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:640:888)
	at t._invokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:640:554)
	at t._receiveRequest (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:639:126)
	at t._receiveOneMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:638:59)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:635:926
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:98:138
	at e.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:99:496)
	at a (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:154:213)
	at Socket._socketDataListener (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:154:432)
	at emitOne (events.js:116:13)
	at Socket.emit (events.js:211:7)
	at addChunk (_stream_readable.js:263:12)
	at readableAddChunk (_stream_readable.js:250:11)
	at Socket.Readable.push (_stream_readable.js:208:10)
	at Pipe.onread (net.js:594:20)

**Comment by vincaslt** [comment]
Hi, thanks, would you be able to provide the full code example or the excerpt where this error occurs?

**Comment by rutenio** [comment]
This is just an example, but it appears in several files that mix javascript, html and php code.
>                                 $.each(data, function(i, v) {
                                    lista += '<tr><td class="text-center">' + v.idconfig +
                                        '</td><td>' + v.titulo + '</td><td class="text-center">' +
                                        '<input class="form-check-input" type="checkbox" disabled ' + (v.ativo == 1 ? 'checked' : '')  +
                                        '></td><td class="text-center">' +
                                        '<button class="btedit btn btn-link btn-sm" data-id="' + v.idconfig + '">' +
                                        '<i class="fas fa-edit"></i></button></td></tr>'
                                });
                                $('#tableconfigs tbody').html(lista);

**Comment by vincaslt** [comment]
I have improved the matching according to your example. However, I wouldn't bet on it working well all the time, because building DOM from strings like that is incredibly hard to parse correctly. It's almost an accident if it works, and I will not support it if it requires hacks. Having said that, you can try updating the extension now and check if it works better for you now.

**Comment by rutenio** [comment]
Thank you Vincas. My intention was only that the extension would still work in the rest of the html, and now, besides working in HTML, it still works in the javascript string, incredible!
